### DISCUSSION TITLE
Automatic database purge

### DISCUSSION THREAD

**Discussion by Valeoria** [Zrips/CMI#6636] [Feature Requests]
Currently the "Purge" options in the main config only cover purging of the world folder data.

Would be very useful if we had the option to purge the CMI database itself, completely removing all mention of the player after the "OfflineDays". Currently with the settings listed below, we have players returning after 120 days and their playerdata from the world file is reset correctly, but they still have a balance, homes etc... from CMI. This really interferes with our "120 days inactive will delete your playerdata" policy. We have far too many players to manually use /cmi removeuser, completely out of the question.

I believe a lot of people may be in the same situation and could greatly benefit from this option. This would also keep the filesize low for the database. We are currently at a few gigabytes.

purge:
  CleanOnStart: true
  OfflineDays: 120
  PlayerData:
    Enabled: true
    SourceFolder: survival_117/playerdata
    DeleteFiles: false
    DestinationFolder: survival_117/playerdata_backup
  PlayerStats:
    Enabled: true
    SourceFolder: survival_117/stats
    DeleteFiles: false
    DestinationFolder: survival_117/stats_backup
  PlayerAdvancements:
    Enabled: true
    SourceFolder: survival_117/Advancements
    DeleteFiles: false
    DestinationFolder: survival_117/Advancements_backup

**Comment by CloudeLecaw** [comment]
A purge command would be nice to set.
So we could add something like the cmi removeuser command, that removes abolayer from cmi database
